core/src/main/java/org/apache/calcite/sql/SqlDialect.java

```diff
@@ -148,15 +151,11 @@
 
   /**
    * Valid Date Time Separators.
+   * '@' is a standard Format added at mig side in case if we don't have any separators
+   * between two tokens which makes us easy to getFinalFormat
```

Comment:
also add that these separators should be selected such that they are never a part of any pattern in SqlDateTimeFormat.